notebooks/standardize_events_data.ipynb

```diff
@@ -90,90 +297,914 @@
+    "# standard_zk.head()\n",
+    "\n",
+    "# Uncomment the following below to Save the resulting dataframe to a new CSV file\n",
+    "# output_file_path = 'extracted_standard_zk_events.csv'\n",
+    "# standard_zk.to_csv(output_file_path, index=False)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "0737a215-bd96-49d9-b236-215060d00558",
+   "id": "99932d49-9099-4bed-b81f-e733abf42269",
    "metadata": {},
    "source": [
-    "4. Apply the Transformation Function to the DataFrame\n",
-    "Apply the decode_and_convert_row function to each row of the DataFrame to create a new standardized DataFrame."
+    "### For nostra_alpha protocol, these are the events and their relevant data\n",
+    "- for debt event: user is in data[0], token is in from_address, face_amount is in data[1]\n",
+    "- for mint event: user is in data[0], token is in from_address, face_amount is in [1]\n",
+    "- for interest state updated event :  debtToken is in data[0], lendingRate is in data[1], borrowRate is in data[3], lendIndex is in [5], borrowIndex is in [7]"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "6dc5df33-6f26-4cad-a4a6-9d3c42e64559",
+   "execution_count": 7,
+   "id": "11dc202e-2df0-48e1-9d02-b4384c0412d8",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Apply the function to each row\n",
-    "standardized_rows = df.apply(decode_and_convert_row, axis=1)\n",
-    "standardized_df = pd.DataFrame(standardized_rows.tolist())"
+    "# Define a function to decode bytes, convert to list, and extract specific elements based on the event\n",
+    "def extract_nostra_alpha(row):\n",
+    "    \n",
+    "    def hex_to_int(hex_str):\n",
+    "        return int(hex_str, 16)\n",
+    "    \n",
+    "    try:\n",
+    "        # Decode the bytes to a string and convert the string to an actual list\n",
+    "        data_str = row['data'].decode('utf-8')\n",
+    "        data_list = eval(data_str)\n",
+    "   \n",
+    "        if row['key_name'] == 'Mint':\n",
+    "            return pd.Series({\n",
+    "                'user': data_list[0],\n",
+    "                'token': row['from_address'],\n",
+    "                'face_amount': hex_to_int(data_list[1])\n",
+    "            })\n",
+    "        elif row['key_name'] == 'Debt':\n",
+    "            return pd.Series({\n",
+    "                'user': data_list[0],\n",
+    "                'token': row['from_address'],\n",
+    "                'face_amount': hex_to_int(data_list[1])\n",
+    "            })\n",
+    "        elif row['key_name'] == 'InterestStateUpdated':\n",
+    "            return pd.Series({\n",
+    "                'debt_token': data_list[0],\n",
+    "                'lending_rate': data_list[1],\n",
+    "                'borrow_rate': data_list[3],\n",
+    "                'lend_index': data_list[5],\n",
+    "                'borrow_index': data_list[7]\n",
+    "            })\n",
+    "        else:\n",
```

Comment:
What are the other key names that we ignore? For example, `Burn` events are missing.